### ISSUE TITLE
Getters and setters with computed property names crash the compiler

### ISSUE THREAD

**Issue by sjrd** [google/closure-compiler#3364]
This a regression since v20190415, observed on latest master (9b7948222de73987cceff3e394147b4218a50559).

Here are the steps to reproduce.

First build the mentioned commit with:
```bash
mvn -DskipTests -pl externs/pom.xml,pom-main.xml,pom-main-shaded.xml
```

Create the file `test.js` with:
```js
class Bug {
  get ["member"]() {
    return 42;
  };
}
console.log(new Bug());
```
Then run:
```bash
$ java -jar target/closure-compiler-1.0-SNAPSHOT.jar --js_output_file=out.js -O SIMPLE_OPTIMIZATIONS --language_out ECMASCRIPT_2015 --jscomp_off '*' test.js
```
Observe the following result:
```
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

COMPUTED_PROP 2 [length: 37] [source_file: test.js] [computed_prop_getter: 1] is not a string node
  Node(CLASS_MEMBERS): test.js:1:0
class Bug {
  Parent(CLASS): test.js:1:0
class Bug {

        at com.google.javascript.rhino.Node.getString(Node.java:1234)
        at com.google.javascript.jscomp.DeadPropertyAssignmentElimination$GetterSetterCollector.visit(DeadPropertyAssignmentElimination.java:529)
        at com.google.javascript.jscomp.NodeTraversal.traverseClassMembers(NodeTraversal.java:965)
        at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:859)
        at com.google.javascript.jscomp.NodeTraversal.traverseClass(NodeTraversal.java:951)
        at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:857)
        at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:979)
        at com.google.javascript.jscomp.NodeTraversal.handleScript(NodeTraversal.java:804)
        at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:839)
        at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:979)
        at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:863)
        at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:371)
        at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:381)
        at com.google.javascript.jscomp.DeadPropertyAssignmentElimination.process(DeadPropertyAssignmentElimination.java:80)
        at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:328)
        at com.google.javascript.jscomp.PhaseOptimizer$Loop.process(PhaseOptimizer.java:473)
        at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:237)
        at com.google.javascript.jscomp.Compiler.performOptimizations(Compiler.java:2380)
        at com.google.javascript.jscomp.Compiler.lambda$stage2Passes$1(Compiler.java:780)
        at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:102)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.UnsupportedOperationException: COMPUTED_PROP 2 [length: 37] [source_file: test.js] [computed_prop_getter: 1] is not a string node
        ... 24 more
```
The issue cannot be reproduced with v20190415. It also cannot be reproduced with WHITESPACE_ONLY (it is only triggered by the simple or advanced optimizations).

The issue *can* also be reproduced with a setter instead of a getter.

`git bisect` says that the culprit is d52df21eb3dbd31d47d385de38e72ce5327745b3. `git revert`ing that commit on top of master fixes the issue.

**Comment by brad4d** [comment]
Created internal issue http://b/132178796

@sjrd,  thanks a lot for identifying the breaking commit!

**Comment by EatingW** [comment]
Fixed by 4ff08d2

**Comment by sjrd** [comment]
Thank you!
